### PR TITLE
Problem in fixup_pth_file() - second cut :-)

### PR DESCRIPTION
**Pull request by JAdata**
This is a revamp of Pull Request #32 with a better commit sequence and the changing of the version number removed.

### CODE REVIEW

**Comment by timabbott** [comment]
@umairwaheed can you review + test this?

**Comment by umairwaheed** [comment]
Sure.

File: tests/test_fixup_scripts.py [hunk 1]
```diff
@@ -187,3 +190,28 @@ def test_fixup_script_different_version(self):
...
+
+        content = os.linesep.join(('/usr/local/bin/someFile', '/tmp/xyzzy', '/usr/local/someOtherFile', ''))
+        pth = '/tmp/test_fixup_pth_file.pth'
+
+        with open(pth, 'w') as fd:
+            fd.write(content)
+
+        # Find size of file - should be plain ASCII/UTF-8...
+
+        orgStat = os.stat(pth)
```
Comment (umairwaheed) [inline tests/test_fixup_scripts.py]: Can you please change `orgStat` to `stat_result`?

File: tests/test_fixup_scripts.py [hunk 2]
```diff
@@ -187,3 +190,28 @@ def test_fixup_script_different_version(self):
...
+        assert len(content) == orgStat[stat.ST_SIZE]
+
+        # Now sleep for 2 seconds, then call fixup_pth_file(). This should ensure that the stat.ST_MTIME has
+        # changed if the file has been changed/rewritten
+
+        time.sleep(2)
+        clonevirtualenv.fixup_pth_file(pth, '/usr/local', '/usr/xyzzy')
+        newStat = os.stat(pth)
```
Comment (umairwaheed) [inline tests/test_fixup_scripts.py]: Can you please change to`newStat` to `new_stat_result`?

File: clonevirtualenv.py [hunk 3]
```diff
@@ -108,7 +109,7 @@ def fix_symlink_if_necessary(src_dir, dst_dir):
                 target = os.path.realpath(full_file_path)
                 if target.startswith(src_dir):
                     new_target = target.replace(src_dir, dst_dir)
-                    logger.debug('fixing symlink in {}'.format(full_file_path))
+                    logger.debug('fixing symlink in %s' % full_file_path)
```
Comment (umairwaheed) [inline clonevirtualenv.py]: Why is this needed?

File: clonevirtualenv.py [hunk 4]
```diff
@@ -298,7 +305,7 @@ def main():
     try:
         old_dir, new_dir = args
     except ValueError:
-        print("virtualenv-clone {}".format(__version__))
+        print("virtualenv-clone %s" % __version__)
```
Comment (umairwaheed) [inline clonevirtualenv.py]: This change shouldn't be required.

File: clonevirtualenv.py (hunk 3 above)
Comment (JAdata) [reply clonevirtualenv.py]: Originally the existing tox.ini file included testing python2.6. 2.6 did not accept empty format directives '{}', so I rewrote them in "portable", 2/3 independent way.  This way it also works for 2.6 even if I later changes the tox.ini to match the travis config which does not test 2.6... This is perfectly legal and accepted python 3...

File: clonevirtualenv.py (hunk 4 above)
Comment (JAdata) [reply clonevirtualenv.py]: Same reason as the above "2.6 failing" story

File: tests/test_fixup_scripts.py (hunk 1 above)
Comment (JAdata) [reply tests/test_fixup_scripts.py]: I will change from camelCase to snake_case...:-)

**Comment by JAdata** [comment]
Ooops, something is wrong with yor Travis setup - it says python 3.3 and 3.4 NOT INSTALLED. This obviously worked when I created the PR. You fix?

**Comment by timabbott** [comment]
Those failures are Travis itself flaking; I just restarted the 2 jobs that failed on startup.

**Comment by timabbott** [comment]
On the front of `%s` vs. `format`, I think it's reasonable to switch to `%s`, though I'd have put those in their own commit explaining it's to make Python 2.6 work.  Is that all one needs to have it run with Python 2.6?

**Comment by JAdata** [comment]
Yes, that is all that was needed to make it run under 2.6. No, I will not redo this yet another time (redo the commits) just for a comment to be more consistent. Let's not loose track of the goal here, to fix a specific problem. Nobody will ever care whether this was one or two commits. Sorry, it's late and I got a little frustrated dealing with what I feel are irrelevant details...

**Comment by timabbott** [comment]
Great.  I think those details are important for people looking to see if a bug they care about is fixed here, so took the 3 minutes to clean up the commit history as I described with `git rebase` and merged this, thanks @JAdata!

**Comment by timabbott** [comment]
@JAdata Do you think we should add 2.6 back to the tox.ini and in Travis?

**Comment by JAdata** [comment]
Well, since "Python 2.6 is no longer supported by the Python core team", I think I would drop it. It IS p.t., with my '{} fix', 2.6 compliant and probably will be until you do a major rewrite, but at some point you must "let go" :-)
